build/yamls/antrea-windows-ovs-containerd.yml

```diff
@@ -0,0 +1,331 @@
+apiVersion: v1
+data:
+  Install-WindowsCNI-Containerd.ps1: |
+    $ErrorActionPreference = "Stop";    
+    mkdir -force c:/var/log/antrea
+
+    $mountPath = $env:CONTAINER_SANDBOX_MOUNT_POINT
+    $mountPath =  ($mountPath.Replace('\', '/')).TrimEnd('/') 
+    mkdir -force C:/var/run/secrets/kubernetes.io/serviceaccount
+    cp $mountPath/var/run/secrets/kubernetes.io/serviceaccount/ca.crt C:/var/run/secrets/kubernetes.io/serviceaccount
+    cp $mountPath/var/run/secrets/kubernetes.io/serviceaccount/token C:/var/run/secrets/kubernetes.io/serviceaccount
+    mkdir -force c:/opt/cni/bin/
+    mkdir -force c:/etc/cni/net.d/
+    cp $mountPath/k/antrea/cni/* c:/opt/cni/bin/
+    cp $mountPath/etc/antrea/antrea-cni.conflist c:/etc/cni/net.d/10-antrea.conflist
+    mkdir -force c:/k/antrea/bin
+    cp $mountPath/k/antrea/bin/antctl.exe c:/k/antrea/bin/antctl.exe
+    
+  Run-AntreaAgent-Containerd.ps1: |
+    $ErrorActionPreference = "Stop"
+    $mountPath = $env:CONTAINER_SANDBOX_MOUNT_POINT
+    $mountPath =  ($mountPath.Replace('\', '/')).TrimEnd('/')
+    & "$mountPath/k/antrea/bin/antrea-agent.exe" --config=$mountPath/etc/antrea/antrea-agent.conf --logtostderr=false --log_dir=c:/var/log/antrea --alsologtostderr --log_file_max_size=100 --log_file_max_num=4 --v=0
+
+  Run-AntreaOVS-Containerd.ps1: |
+    $ErrorActionPreference = "Stop"
+    $OVSInstallDir = "C:\openvswitch"
+    $OVS_DB_SCHEMA_PATH = "$OVSInstallDir\usr\share\openvswitch\vswitch.ovsschema"
+    $OVS_DB_PATH = "$OVSInstallDir\etc\openvswitch\conf.db"
+    if ($(Test-Path $OVS_DB_SCHEMA_PATH) -and !$(Test-Path $OVS_DB_PATH)) {
+        Set-Location -Path "C:\openvswitch\usr\bin
+        ovsdb-tool create "$OVS_DB_PATH" "$OVS_DB_SCHEMA_PATH"
+    }
+    $mountPath = $env:CONTAINER_SANDBOX_MOUNT_POINT
+    $mountPath = ($mountPath.Replace('\', '/')).TrimEnd('/')
+    [Environment]::SetEnvironmentVariable("PATH", "$([Environment]::GetEnvironmentVariable('PATH', 'Machine'));$mountPath/openvswitch/usr/sbin;$mountPath/openvswitch/usr/bin", "Machine")
+    ovsdb-server $OVS_DB_PATH -vfile:info --remote=punix:db.sock --remote=ptcp:6640 --log-file=/var/log/antrea/openvswitch/ovsdb-server.log --pidfile --detach
+    ovs-vsctl --no-wait init
+
+    # Set OVS version.
+    $OVS_VERSION=$(Get-Item $OVSInstallDir\driver\OVSExt.sys).VersionInfo.ProductVersion
+    ovs-vsctl --no-wait set Open_vSwitch . ovs_version=$OVS_VERSION
+
+    ovs-vswitchd --log-file=/var/log/antrea/openvswitch/ovs-vswitchd.log --pidfile -vfile:info --detach
+
+    $SleepInterval = 30
+    while ($true) {
```

Comment:
shall we follow Linux antrea-ovs script https://github.com/antrea-io/antrea/blob/main/build/images/scripts/start_ovs#L160

Reply:
A related question is how to notify the container to quit when the Pod is deleted if using Sleep?

Reply:
@wenyingd @rajnkamr , I tried few things on windows and here are the observations.
1. antrea-agent Pod is running as a daemonset, so just deleting the agent Pod recreates a new agent Pod and in this process OVS processes also exit and start again. I confirmed this by process IDs of OVS processes before deleting agent Pod and after creation of new agent Pod. OVS Process IDs were different both time.
2.  antrea-agent Pod is running on windows and OVS processes are also running fine, now i exit the OVS process from windows Node, it automatically gets started again within 30 seconds as we have written script for it.
3. Now if i remove antrea using `kubectl delete -f antrea-windows-ovs-containerd.yml` then agent Pod is removed and OVS processes also exit and OVS processes doesn't start again. I checked this manually for more than 30 seconds.

Reply:
> antrea-agent Pod is running on windows and OVS processes are also running fine, now i exit the OVS process from windows Node, it automatically gets started again within 30 seconds as we have written script for it.

This is not as expected to restart OVS in 30s when something is wrong. Think this case, OVS is crash and wait to start until 30s, then antrea-agent is not able to connect to OVS during this time. Then the container of antrea-agent would restart again and again. The Pod can not go into running status for 30s?

Reply:
@Atish-iaf @wenyingd , 
On Linux node, ovs status is checked every 30s with CHECK_OVS_INTERVAL=30, we can take a call if we want to reduce this duration to say 15s on window node for better availability of ovs processes in case it crashed.

Reply:
> A related question is how to notify the container to quit when the Pod is deleted if using Sleep?

@wenyingd  Does it get answered by this comment ?

> Now if i remove antrea using kubectl delete -f antrea-windows-ovs-containerd.yml then agent Pod is removed and OVS processes also exit and OVS processes doesn't start again. I checked this manually for more than 30 seconds from windows Node.
I think Start-Sleep exits after getting terminate signal even if it is in sleep interval unlike linux sleep.

Reply:
> > antrea-agent Pod is running on windows and OVS processes are also running fine, now i exit the OVS process from windows Node, it automatically gets started again within 30 seconds as we have written script for it.
> 
> This is not as expected to restart OVS in 30s when something is wrong. Think this case, OVS is crash and wait to start until 30s, then antrea-agent is not able to connect to OVS during this time. Then the container of antrea-agent would restart again and again. The Pod can not go into running status for 30s?

So, here do we expect to restart OVS process earlier than 30 secs or exit the script and quit container ?

Reply:
> So, here do we expect to restart OVS process earlier than 30 secs or exit the script and quit container ?

Yes.